src/containers/EvmCrossChainRecoveryCoin/EvmCrossChainRecoveryCoin.tsx

```diff
@@ -0,0 +1,310 @@
+import { useNavigate, useParams } from 'react-router-dom';
+import { CoinsSelectAutocomplete } from '~/components';
+import { useAlertBanner } from '~/contexts';
+import {
+  assert,
+  isRecoveryTransaction,
+  safeEnv,
+  toWei,
+} from '~/helpers';
+import { BackToHomeHelperText } from '~/components/BackToHomeHelperText';
+import { evmCrossChainRecoveryCoins } from '~/helpers/config';
+import { PolygonFormColdWallet } from './PolygonFormColdWallet';
+import { PolygonFormHotWallet } from './PolygonFormHotWallet';
+import { PolygonFormCustodyWallet } from './PolygonFormCustodyWallet';
+
+async function isDerivationPath(id: string, description: string) {
+  if (id.length > 2 && id.indexOf('m/') === 0) {
+    const response = await window.commands.showMessageBox({
+      type: 'question',
+      buttons: ['Derivation Path', 'Seed'],
+      title: 'Derivation Path?',
+      message: `Is the provided value a Derivation Path or a Seed?\n${description}: ${id}\n`,
+    });
+    return !!response.response;
+  }
+  return false;
+}
+
+function getHotWalletCoinForm() {
+  const { env, coin } = useParams<'env' | 'coin'>();
+  const bitGoEnvironment = safeEnv(env);
+  const [, setAlert] = useAlertBanner();
+
+  const navigate = useNavigate();
+
+  switch (coin) {
+    case 'polygon':
+    case 'tpolygon':
+      return (
+        <PolygonFormHotWallet
```

Comment:
In the future if we add a new coin, do we need to create a new component for it?